### PR TITLE
Add interned s2n-tls+aws-lc test and re-factor script

### PR DESCRIPTION
**Pull request by torben-hansen**
### Issues:

CryptoAlg-1088

### Description of changes: 

This PR accomplishes two things:

Firstly, interned `s2n-tls+aws-lc` is a thing now https://github.com/aws/s2n-tls/issues/3262. `s2n-tls` has a complete and comprehensive set of tests. But we ought to have minimal assurance we don't regress making the feedback loop for trivial errors shorter. Hence, we add a few trivial intern tests to the CI.

Secondly, generally re-factors the `s2n-tls` integrations tests to remove code re-use. Also, run `s2n-tls` unit tests for all build flavors, not only the shared flavor.

This PR will affect the running time of the `s2n-tls` integration tests. In the existing test only one execution of the `s2n-tls` unit tests is performed. This PR will execute the unit tests a total of four times and add two extra `s2n-tls` builds. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by torben-hansen** [comment]
Blocked on https://github.com/aws/s2n-tls/issues/3308

File: tests/ci/run_s2n_integration.sh
Comment (torben-hansen) [inline tests/ci/run_s2n_integration.sh:55]: Need to change this before merging. But not before https://github.com/aws/s2n-tls/pull/3346 is merged

**Review by bryce-shang — DISMISSED** [review]

File: tests/ci/run_s2n_integration.sh
Comment (torben-hansen) [reply tests/ci/run_s2n_integration.sh]: 👍 makes sense

File: tests/ci/run_s2n_integration.sh
Comment (torben-hansen) [reply tests/ci/run_s2n_integration.sh]: Fix merged https://github.com/aws/s2n-tls/commit/36932b2875c97495e6b41c9b1f1e9b43a6e4bf19

So, pointing this to the tip of `s2n-tls` main branch.

File: tests/ci/run_s2n_integration.sh
Comment (torben-hansen) [inline tests/ci/run_s2n_integration.sh:38]: Jotted down a tasks to make this more general. Also need to fix `cmake` vs `cmake` on AL2...

https://github.com/awslabs/aws-lc/blob/main/tests/ci/common_posix_setup.sh#L37 handles this. But the build in that function is too specific for this script.

**Review by bryce-shang — APPROVED** [review]

**Review by samuel40791765 — APPROVED** [review]

**Pull request merged**
